### PR TITLE
Repair edit properties in current task [2.x]

### PR DESCRIPTION
**Pull request by matthias-ronge**
Fixes issue #2817

### CODE REVIEW

File: Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java [hunk 1]
```diff
@@ -1288,13 +1290,29 @@ public void setBatchHelper(BatchStepHelper batchHelper) {
...
+                    || !getOccupiedProperties().contains(pp.getName()))) {
                 answer.add(pp);
             }
         }
         return answer;
     }
 
+    private Set<String> getOccupiedProperties() {
+        Set<String> propertiesMitWerten = new HashSet<>();
+        for (ProcessProperty pp : processPropertyList)
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java]: Please use bracket here in in the next lines.

File: Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java [hunk 2]
```diff
@@ -1288,13 +1290,29 @@ public void setBatchHelper(BatchStepHelper batchHelper) {
...
+            if (pp.getContainer() == 0 && (isEmptyString(pp.getValue())
+                    || !getOccupiedProperties().contains(pp.getName()))) {
                 answer.add(pp);
             }
         }
         return answer;
     }
 
+    private Set<String> getOccupiedProperties() {
+        Set<String> propertiesMitWerten = new HashSet<>();
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java]: German / English mixed variable name, please change it to English.

File: Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java [hunk 3]
```diff
@@ -1288,13 +1290,29 @@ public void setBatchHelper(BatchStepHelper batchHelper) {
...
+        for (ProcessProperty pp : processPropertyList)
+            if (isEmptyString(pp.getValue())) propertiesMitWerten.add(pp.getName());
+        return propertiesMitWerten;
+    }
+
+    /**
+     * @param theString the string
+     * @return true or false
+     */
+    private boolean isEmptyString(String theString) {
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java]: You are testing for non empty value but you are naming the method for test of empty value. You should adjust the test or method name.

File: Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java [hunk 4]
```diff
@@ -1288,13 +1290,29 @@ public void setBatchHelper(BatchStepHelper batchHelper) {
...
+        return theString != null && !theString.isEmpty();
```
Comment (Beacze) [inline Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java]: Is it necessary to write own method for it? Is here not available StringUtils.isBlank(value)?

Comment (matthias-ronge) [reply Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java]: I did not know that yet! It’s amazing!

File: Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java (hunk 3 above)
Comment (matthias-ronge) [reply Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java]: Has done with it …

File: Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java (hunk 1 above)
Comment (matthias-ronge) [reply Goobi/src/de/sub/goobi/forms/AktuelleSchritteForm.java]: Blocks were added.

**Pull request merged**